src/SFA.DAS.AssessorService.Application.Api/Services/OrganisationMergingService.cs

```diff
@@ -0,0 +1,280 @@
+
+            CreateStandardsSnapshot(mergeOrganisation, primaryOrganisation.EndPointAssessorOrganisationId, "Before");
+            CreateStandardsSnapshot(mergeOrganisation, secondaryOrganisation.EndPointAssessorOrganisationId, "Before");
+
+            // Perform the merge.
+
+            MergeOrganisationStandardsAndVersions(
+                primaryOrganisation,
+                secondaryOrganisation,
+                mergedByUserId,
+                secondaryStandardsEffectiveTo);
+
+            // Create the "After" snapshot.
+
+            CreateStandardsSnapshot(mergeOrganisation, primaryOrganisation.EndPointAssessorOrganisationId, "After");
+            CreateStandardsSnapshot(mergeOrganisation, secondaryOrganisation.EndPointAssessorOrganisationId, "After");
+
+            // check
+            /*
+            List<Object> modifiedOrAddedEntities = _dbContext.ChangeTracker.Entries()
+                .Where(x => x.State == Microsoft.EntityFrameworkCore.EntityState.Added
+                    || x.State == Microsoft.EntityFrameworkCore.EntityState.Modified
+                    )
+                .Select(x => x.Entity).ToList();
+            */
+
+            // Now save all the changes.
+            try
+            {
+                await _dbContext.SaveChangesAsync();
+            }
+            catch(Exception ex)
+            {
+                // check rollback has worked
+                throw ex;
+            }
+
+            return mergeOrganisation;
+        }
+
+        private MergeOrganisation CreateMergeOrganisations(string primaryEndpointAssessorOrganisationId, string secondaryEndpointAssessorOrganisationId, Guid createdByUserId)
+        {
+            var mo = new MergeOrganisation()
+            {
+                PrimaryEndPointAssessorOrganisationId = primaryEndpointAssessorOrganisationId,
+                SecondaryEndPointAssessorOrganisationId = secondaryEndpointAssessorOrganisationId,
+                CreatedBy = createdByUserId,
+                CreatedAt = DateTime.UtcNow,
+                ApprovedBy = createdByUserId,
+                ApprovedAt = DateTime.UtcNow,
+                Status = MergeOrganisationStatus.Approved,  // auto-approved for now
+            };
+            _dbContext.MergeOrganisations.Add(mo);
+
+            return mo;
+        }
+
+        private void CreateStandardsSnapshot(MergeOrganisation mo, string endpointAssessorOrganisationId, string replicates)
+        {
+            var organisationStandards = _dbContext.OrganisationStandard.Where(e => e.EndPointAssessorOrganisationId == endpointAssessorOrganisationId);
+            foreach (var os in organisationStandards)
+            {
+                mo.MergeOrganisationStandards.Add(CreateMergeOrganisationStandard(replicates, os));
+
+                var organisationStandardVersions = _dbContext.OrganisationStandardVersion.Where(e => e.OrganisationStandardId == os.Id);
+                foreach (var osv in organisationStandardVersions)
```

Comment:
Which if you tweak the above means you can streamline this to:
`os.OrganisationStandardVersions.Select(x => CreateMergeOrganisationStandardVersion(replicates, x));`

And similiar to delivery areas